lib/iris/coords.py

```diff
@@ -1391,6 +1394,9 @@ def __eq__(self, other):
 
     # The __ne__ operator from Coord implements the not __eq__ method.
 
+    def __hash__(self):
+        return hash(id(self))
```

Comment:
IIRC, the "mutable-hashable" was originally a cheap/lazy hack to simplify the cube-merge code. During the merge process the cubes/coords don't change so they are effectively immutable. Unsurprisingly, usage of the "mutable-hashable" behaviour has proliferated (e.g. `iris.iterate.izip`).

Ideally we should get rid of mutable-hashable behaviour, but not in this PR.